geps/gep-1742.md

```diff
@@ -301,8 +301,87 @@ Could not find any HTTP specific timeouts. PRs welcomed. 😊
 
 ## API
 
-TBD.
+From the above analysis, it appears that most implementations are capable of supporting
+the configuration of simple downstream request timeouts on HTTPRoute rules. This is a
+relatively small addition (one new field) that would benefit many users.
+
+Because request timeouts are very much related to retry attempts, we also propose
+addressing issue: [#1731](https://github.com/kubernetes-sigs/gateway-api/issues/1731)
+by adding another field for retry conifguration in an HTTPRoute rule. This would also
+benefit many users and seems to be implementable by many (most?) Gateway implementations.
+
+### GO
+
+```go
+type HTTPRouteRule struct {
+	// Timeout for HTTP requests, disabled by default.
+	Timeout *time.Duration `json:"timeout,omitempty"`
+	// Retry policy for HTTP requests.
+	Retries *HTTPRetry `json:"retries,omitempty"`
+  // ...
+}
+
+type HTTPRetry struct {
+	// Number of retries to be allowed for a given request.
+	Attempts int32 `json:"attempts,omitempty"`
+	// Timeout per attempt for a given request, including the initial call and any retries.
+	// Default is the value of the HTTPRouteRule request Timeout.
+	PerTryTimeout *time.Duration `json:"per_try_timeout,omitempty"`
+	// Specifies the conditions under which retry takes place.
+	// One or more conditions can be specified using a ‘,’ delimited list.
+	RetryOn string `json:"retry_on,omitempty"`
+	// Flag to specify whether the retries should retry to other localities.
+	RetryRemoteLocalities *bool `json:"retry_remote_localities,omitempty"`
+}
+```
+
+### YAML
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: timeout-example
+spec:
+  ...
+  rules:
+  - backendRefs:
+    - name: some-service
+      port: 8080
+    timeout: 10s
+    retries:
+      attempts: 3
+      perTryTimeout: 2s
+      retryOn: 503
+```
+
+### Timeout values
+
+Timeout and PerTryTimeout values are formatted as 1h/1m/1s/1ms and MUST BE >= 1ms.
```

Comment:
Is this used anywhere else in gateway-api or core k8s? Is there a more formal spec?

is 1h5m valid?

Reply:
I couldn't find any other use of duration in gateway-api. I couldn't find it in core k8s either, but maybe not looking in the right place? Otherwise, I think this is just the Go time.Duration string serialization that we'd use?